src/test/java/br/com/academiadev/reembolsoazul/reembolsoazul/ReembolsoazulApplicationTests.java

```diff
@@ -188,5 +192,34 @@ public void companyCodeGeneration() {
 
 		Assert.assertTrue(!hash1.equals(hash2));
 	}
+	
+	@Test
+	public void registerCompanyAndUser() {
+		// register company 1
+		UserCompanyRegisterDTO userCompanyRegisterDTO = new UserCompanyRegisterDTO();
+		userCompanyRegisterDTO.setName("Pessoa 1");
+		userCompanyRegisterDTO.setEmail("email@pessoa.com");
+		userCompanyRegisterDTO.setPassword("123456");
+		userCompanyRegisterDTO.setCompanyName("Empresa legal");
+		
+		try {
+			userController.registerUserCompany(userCompanyRegisterDTO);
+		} catch (CompanyNotFoundException e) {
+			e.printStackTrace();
+		}
+		
+		//Find the company and user registered
+		Company company = companyRepository.findByName("Empresa legal").get(0);
+		User user = userRepository.findByEmail("email@pessoa.com");
+		
+		Assert.assertTrue(
+				user.getName().equals("Pessoa 1") &&
+				user.getEmail().equals("email@pessoa.com") &&
+				passwordEncoder.matches("123456", user.getPassword()) &&
+				user.getCompany().getName().equals("Empresa legal") &&
+				user.getUserType() == UserType.ROLE_ADMIN &&
+				company.getName().equals("Empresa legal")
+		);
+	}
 
```

Comment:
It would be nice to split this test file later on, so each test file have responsibility over a specific part of the backend.